### PR TITLE
docs: navigation structure with embedded tutorial pages

### LINKED ISSUE #894: [OpenSTEF 4.0] Documentation navigation structure and tutorial pages
## Context

The documentation site needs an expandable navigation structure with embedded tutorial notebooks that appear in both the Examples section and the User Guide.

**Parent**: #883

## Goal

Set up the docs navigation skeleton with card-based landing pages, embedded tutorial notebooks in the user guide, and proper sidebar context for each section.

## Tasks

- [x] Create expandable navigation with card layouts for Getting Started, Guides, and Concepts
- [x] Write comprehensive Feature Engineering tutorial covering all 25+ transforms
- [x] Embed tutorials in user guide via `docs/sync_sources.py` (separate Sphinx documents for correct sidebar)
- [x] Export `LagsAdder` and `VersionedLagsAdder` for autodoc API page generation
- [x] Add explicit nav titles for clean sidebar display
- [ ] Add remaining user guide content pages (forecasting, datasets, deployment, etc.)
- [ ] Embed additional tutorials in relevant user guide sections as they are written

## Constraints

- Notebooks must execute in < 60s (uses `nb_execution_mode = "cache"`)
- Each embedded tutorial is a separate Sphinx document to avoid cross-toctree sidebar issues
- `docs/sync_sources.py` runs as `_docs_sync` poe task before every build

## Acceptance Criteria

- Navigation renders correctly with proper sidebar context in both Examples and User Guide
- Feature Engineering tutorial covers all transform groups with interactive plots
- `poe all` passes
- `poe docs` builds without errors

### PR DESCRIPTION
**Pull request by egordm**
Closes #894

## Summary

Sets up the documentation navigation structure with expandable nav, card layouts, and embedded tutorial pages.

### Changes

**Feature Engineering Tutorial** (comprehensive rewrite):
- Covers all 25+ transforms across 6 groups (time, weather, energy, general, validation, postprocessing)
- Interactive Plotly visualizations for each transform
- API cross-references (\`{class}\` links) throughout
- Custom transform example (RateOfChangeAdder)
- Pipeline composition section
- Summary reference table

**Navigation & Structure:**
- Embed tutorials in user guide via \`docs/sync_sources.py\` (separate Sphinx documents = correct sidebar context)
- Explicit nav titles for clean left-sidebar display
- Card-based index pages for Getting Started and Guides sections

**API Docs:**
- Export \`LagsAdder\` and \`VersionedLagsAdder\` in \`time_domain/__init__.py\` so autodoc generates their API pages

**Build:**
- New \`docs/sync_sources.py\` script (replaces inline \`_docs_sync\` snippet)
- Gitignore generated tutorial copies

### Testing
- \`poe all\` passes (lint, format, tests, jupytext sync)
- Notebook executes in ~40s
- Docs build succeeds (210 warnings, all pre-existing)

### CODE REVIEW

File: examples/tutorials/feature_engineering.py
Comment (lschilders) [inline examples/tutorials/feature_engineering.py:1]: Same comment as in other PR: call `features_added()` rather than manual check for DataFrame columns

File: examples/tutorials/feature_engineering.py [hunk 1]
```diff
@@ -0,0 +1,1041 @@
...
+result = lags.transform(sample)
+
+lag_cols = [c for c in result.data.columns if c not in sample.data.columns]
+print(f"Added {len(lag_cols)} lag columns: {lag_cols[:5]}")
+
+# %% tags=["hide-input"]
+# Plot load vs its 7-day lag
+one_week = result.data.iloc[672:1344]
+lag_7d_col = [c for c in lag_cols if "7 day" in c.lower() or "168" in c]
+target_col = lag_7d_col[0] if lag_7d_col else lag_cols[0]
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: Might be nicer to plot 2 weeks here rather than 1, to better visualize the weekly repetition.

File: examples/tutorials/feature_engineering.py [hunk 2]
```diff
@@ -0,0 +1,1041 @@
...
+
+# %% tags=["hide-input"]
+fig = make_subplots(
+    rows=2,
+    cols=1,
+    shared_xaxes=True,
+    vertical_spacing=0.08,
+    subplot_titles=("Daily Load Envelope", "Daylight Duration"),
+)
+daily = result.data["load"].resample("D")
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: Not really sure what the load in the plot should be telling, does not seem to be affected by the daylight_continuous. Might be fine to plot the feature solely, maybe even over the whole year?

File: examples/tutorials/feature_engineering.py [hunk 3]
```diff
@@ -0,0 +1,1041 @@
...
+# - **Dewpoint** → discomfort threshold → HVAC demand
+# - **Air density** → affects wind turbine output
+# - **Humidity** → reduces PV efficiency
+
+# %%
+from openstef_models.transforms.weather_domain import AtmosphereDerivedFeaturesAdder
+
+atmosphere = AtmosphereDerivedFeaturesAdder(
+    included_features=["dewpoint", "air_density"],
+    temperature_column="temperature_2m",
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py:466]: Align the summed up features to `included_features`, `vapour_pressure` is now missing

File: examples/tutorials/feature_engineering.py [hunk 4]
```diff
@@ -0,0 +1,1041 @@
...
+    go.Scatter(
+        x=result.data.index[:672],
+        y=result.data["wind_power"].iloc[:672],
+        mode="lines",
+        name="Power",
+        line={"width": 1.5, "color": "green"},
+    ),
+    row=1,
+    col=2,
+)
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py:570]: Maybe nicer to plot the real curve here, so linearly increasing wind speed and wind power fit, or wind speed at x-axis, so it is more clear what it does.

File: examples/tutorials/feature_engineering.py [hunk 5]
```diff
@@ -0,0 +1,1041 @@
...
+# Simulate 2% missing data
+noisy = sample.data.copy()
+rng = np.random.default_rng(42)
+mask = rng.random(noisy.shape) < 0.02
+noisy[mask] = np.nan
+noisy_ds = TimeSeriesDataset(data=noisy, sample_interval=sample.sample_interval)
+
+result = imputer.fit_transform(noisy_ds)
+print(f"Before: {noisy_ds.data.isna().sum().sum():,} NaN values")
+print(f"After:  {result.data.isna().sum().sum():,} NaN values (load preserved)")
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: Better to print NaNs per column before and after to show load is unaffected

File: examples/tutorials/feature_engineering.py [hunk 6]
```diff
@@ -0,0 +1,1041 @@
...
+
+# %% [markdown]
+# ### {class}`~openstef_models.transforms.general.Scaler`
+#
+# **What:** Standardizes features to zero mean / unit variance.
+#
+# **Why:** Features on different scales (temperature 0-16C vs radiation 0-633 W/m2)
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: Units should be formatted better

File: examples/tutorials/feature_engineering.py [hunk 7]
```diff
@@ -0,0 +1,1041 @@
...
+# **Why:** Peak load forecast errors are costlier (grid congestion, reserve
+# activation). Weighting peaks higher makes the model pay more attention to
+# the periods that matter most operationally.
+
+# %%
+from openstef_models.transforms.general import SampleWeighter
+
+weighter = SampleWeighter()
+result = weighter.fit_transform(sample)
+
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py:672]: Might be nice to print or visualize different sample weighting methods in short, since they matter quite a lot.

File: examples/tutorials/feature_engineering.py [hunk 8]
```diff
@@ -0,0 +1,1041 @@
...
+print("✓ Complete data passes (threshold=0.8)")
+
+# Incomplete data fails
+sparse = sample.data.copy()
+sparse.iloc[:2000, sparse.columns.get_loc("temperature_2m")] = np.nan
+sparse_ds = TimeSeriesDataset(data=sparse, sample_interval=sample.sample_interval)
+
+try:
+    checker.transform(sparse_ds)
+except (InsufficientlyCompleteError, MissingColumnsError) as e:
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: This is a bit of a weird catch, since we just want to show the InsufficientlyCompleteError

File: examples/tutorials/feature_engineering.py [hunk 9]
```diff
@@ -0,0 +1,1041 @@
...
+print("✓ Normal data passes")
+
+# Inject a 12-hour flatline at end
+flat_data = sample.data.copy()
+flat_data.iloc[-48:, flat_data.columns.get_loc("load")] = 300000.0
+flat_ds = TimeSeriesDataset(data=flat_data, sample_interval=sample.sample_interval)
+
+try:
+    flatliner.transform(flat_ds)
+except (FlatlinerDetectedError, MissingColumnsError) as e:
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: Same as previous comment

File: examples/tutorials/feature_engineering.py [hunk 10]
```diff
@@ -0,0 +1,1041 @@
...
+# These operate on {class}`~openstef_core.datasets.ForecastDataset` (after
+# prediction), not training data. They refine model outputs:
+#
+# | Transform | What it does | API Reference |
+# |-----------|-------------|---------------|
+# | {class}`~openstef_models.transforms.postprocessing.ConfidenceIntervalApplicator` | Adds prediction intervals from quantile residuals | {class}`~openstef_models.transforms.postprocessing.ConfidenceIntervalApplicator` |
+# | {class}`~openstef_models.transforms.postprocessing.IsotonicQuantileCalibrator` | Calibrates quantiles via isotonic regression | {class}`~openstef_models.transforms.postprocessing.IsotonicQuantileCalibrator` |
+# | {class}`~openstef_models.transforms.postprocessing.QuantileSorter` | Ensures q10 ≤ q50 ≤ q90 | {class}`~openstef_models.transforms.postprocessing.QuantileSorter` |
+#
+# See {doc}`quantile_calibration` for a full walkthrough. Conceptual usage:
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py]: This quantile_calibration link is broken

File: examples/tutorials/feature_engineering.py [hunk 11]
```diff
@@ -0,0 +1,1041 @@
...
+        RateOfChangeAdder(feature="load"),
+        # 3. Data cleaning
+        Imputer(selection=Exclude("load"), imputation_strategy="mean"),
+        Scaler(selection=Exclude("load"), method="standard"),
+    ]
+)
+
+result = pipeline.fit_transform(sample)
+print(
+    f"Input: {sample.data.shape[1]} cols → Output: {result.data.shape[1]} cols (+{result.data.shape[1] - sample.data.shape[1]} features)"
```
Comment (lschilders) [inline examples/tutorials/feature_engineering.py:999]: Would be nice to add a preprocessing and postprocessing pipeline example here, to make the distinction more clear into the bigger picture.

**Review by lschilders — COMMENTED** [review]
Very nice work! Some suggestions for improvement.

File: examples/tutorials/feature_engineering.py
Comment (egordm) [reply examples/tutorials/feature_engineering.py:1]: Fixed.

File: examples/tutorials/feature_engineering.py (hunk 1 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py]: Fixed. Now 2 weeks are plotted.

File: examples/tutorials/feature_engineering.py (hunk 3 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py:466]: Fixed.

File: examples/tutorials/feature_engineering.py (hunk 4 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py:570]: Fixed. Nice one.

File: examples/tutorials/feature_engineering.py (hunk 5 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py]: Good one. Fixed.

File: examples/tutorials/feature_engineering.py (hunk 7 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py:672]: Added. Good one.

File: examples/tutorials/feature_engineering.py (hunk 9 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py]: Fixed!

File: examples/tutorials/feature_engineering.py (hunk 10 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py]: Fixed!

File: examples/tutorials/feature_engineering.py (hunk 11 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py:999]: Added an example for postprocessing.

File: examples/tutorials/feature_engineering.py (hunk 6 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py]: If you mean like text formatting of units. Then fixed. Now (0-16C → 0–16 °C and 0-633 W/m2 → 0–633 W/m²)

File: examples/tutorials/feature_engineering.py (hunk 7 above)
Comment (lschilders) [reply examples/tutorials/feature_engineering.py:672]: Figures overlap, maybe plot them below each other? Or remove the part in brackets?
image

File: examples/tutorials/feature_engineering.py (hunk 10 above)
Comment (lschilders) [reply examples/tutorials/feature_engineering.py]: Formatting of the code here is still weird: 

image

File: examples/tutorials/feature_engineering.py (hunk 7 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py:672]: Fixed!

**Review by lschilders — APPROVED** [review]

File: examples/tutorials/feature_engineering.py (hunk 10 above)
Comment (egordm) [reply examples/tutorials/feature_engineering.py]: This seems to be an old screenshot. Now it's runnable. Fixed.
